test/core/end2end/end2end_test_suites.cc

```diff
@@ -550,7 +550,29 @@ class InsecureFixtureWithPipeForWakeupFd : public InsecureFixture {
 };
 #endif
 
+std::string GetTempDir(){
+#ifdef WIN32
+  // windows temp dir usually exceeds uds length.
+  // std::array<char, MAX_PATH + 1> buff = {};
+  // DWORD len = GetTempPathA(buff.size(), buff.data());
+  // if (len == 0) {
+  //   GPR_ASSERT(false);
+  // }
+  // return std::string(buff.data(), len);
+  std::string temp_dir = "C:/tmp/";
+  if (CreateDirectoryA(temp_dir.c_str(), NULL) == 0 &&
+    ERROR_ALREADY_EXISTS != GetLastError())
+  {
+    GPR_ASSERT(false); // temp dir create failed
```

Comment:
Prefer `grpc_core::Crash(absl::StrCat("Could not create temp dir: ", temp_dir));`